### PR TITLE
Functions section

### PR DESCRIPTION
**Pull request by jotas6**
Content for the section about functions of the workshop

## Status
- [x] Edit README file
- [x] Code examples
- [x]  Instructor notes
- [x] Reference sheet
- [x] Schedule

### CODE REVIEW

**Comment by jotas6** [comment]
I just noticed that I didn't cover functions that modify some of the arguments (e.g `f!(x)`). Would it be a good idea to include this topic? I could fit it in `02-advanced.jl` at the beginning or create a new script for it.

File: 03-functions/01-syntax.jl [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+greet("James")
+greet("Alice")
+
+add(x, y) = x + y
+
+add(2, 3)
+add(1, 10)
+
+f(x) = x^2 # Natural way to define math functions
```
Comment (storopoli) [inline 03-functions/01-syntax.jl:39]: this should be named compact function assignment

File: 03-functions/01-syntax.jl [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+f(2)
+f(3)
+
+## Return multiple values
+function even_greater_than_10(number)
+
```
Comment (storopoli) [inline 03-functions/01-syntax.jl:46]:
```suggestion
```

File: 03-functions/01-syntax.jl [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    even = iseven(number) # A function that we didn't write
+    greater = number > 10
+
+    return even, greater # Return multiple values with return1, return2, return3, ...
+
```
Comment (storopoli) [inline 03-functions/01-syntax.jl:51]:
```suggestion
```

File: 03-functions/01-syntax.jl [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+end
+
+checks = even_greater_than_10(12) # We get a Tuple
+
+# Extract values by indexing the Tuple
+even = checks[1]
+greater = checks[2]
+
+## Tip: compact way to get multiple values
```
Comment (storopoli) [inline 03-functions/01-syntax.jl:60]:
```suggestion
## Tip: useful way to unpack multiple values
```

File: docs/index.md [hunk 5]
```diff
@@ -12,7 +12,7 @@ This workshop is an **introduction to the Julia programming language,** covering
     - Boolean variables
     - Strings
     - Containers
-
+
```
Comment (storopoli) [inline docs/index.md:15]:
```suggestion

```

File: docs/instructors.md [hunk 6]
```diff
@@ -96,6 +96,43 @@ First, show that macros start with `@` and quickly go over the examples. Next, s
 parenthesis (as in a function). The `@time` and `@doc` macros are used in the examples, so it is a good idea to briefly talk about what those macros do. 
 Lastly, show how `@macroexpand` can be used to see what a macro is doing under the hood. 
 
+## Part 3: functions
+
+The code examples for this section of the workshop can be found in the `03-functions` folder.
+
+Start by going over `01-syntax.jl` as this script will cover the basics of defining and calling functions. First, show the standard way
+of creating functions with the `function` keyword, and then show how a function can be called once it has been defined. Use the examples
+to show that a function can have a single argument, multiple arguments, or no arguments at all. Next, go over the single line syntax to
```
Comment (storopoli) [inline docs/instructors.md:105]: Say about the compact assignment form

File: docs/reference.md [hunk 7]
```diff
@@ -63,6 +63,33 @@ be replaced with underscores (`_`).
 - Macros can be called through spaces or with parenthesis (as in functions).
 - `@macroexpand` can be used to examine what a macro is doing under the hood.
 
+### Part 3: functions
+
+- Complex functions can be created with the `function` keyword, while simpler ones can be 
+more conveniently defined using the single line syntax (e.g `f(x) = x^2`).
```
Comment (storopoli) [inline docs/reference.md:69]: compact assignment form

**Review by storopoli — COMMENTED** [review]
some comments

**Comment by jotas6** [comment]
@storopoli I think I have covered the comments. I changed "single line syntax" to "compact assignment form" in all the places I could find it, and removed some whitespace from a function definition.

**Pull request merged**